src/declarativebookmarkmodel.cpp

```diff
@@ -47,12 +51,12 @@ void DeclarativeBookmarkModel::removeBookmark(const QString& url)
         return;
     }
 
-    if (bookmarks.contains(url)) {
-        int index = bookmarkUrls.indexOf(url);
+    if (bookmarkIndexes.keys().contains(url)) {
```

Comment:
What if we'd drop also bookmarkIndexes QMap. And remove the first occurrence by first checking bookmarkIndex(url) if greater than zero then remove it. We'd need a simple function bookmarkIndex function to find the occurrence.

We're always removing first occurrence.